pkg/ottl/parser.go

```diff
@@ -306,6 +337,19 @@ func (s *StatementSequence[K]) Execute(ctx context.Context, tCtx K) error {
 	return nil
 }
 
+func (s *StatementSequence[K]) getTracer() trace.Tracer {
+	s.tracerOnce.Do(func() {
```

Comment:
this is done like this to make sure that the `tracer` is set to avoid nil pointer access -The tracer should be already set if the `StatementSequence` is initialised via its `New()` function, but in the tests it was created directly, without using the `New()` function.

Reply:
I'd prefer the tests be updated to set a noop.Tracer or be calling New.

Reply:
sure, we can also do that - will adapt the PR accordingly